### ISSUE TITLE
Other scripting/outside interfaces?

### ISSUE THREAD

**Issue by jsblake** [TimBurris/SiriusXMPlayer#3]
I just happened across the project so I haven't had time to dig into the code, but I'm curious: does the way this app works with the website lend itself to additional scripting? 

For instance, I have a Streamdeck+ (https://www.elgato.com/en/stream-deck-plus) and would like to configure some dedicated shortcut buttons to start/stop SXM specifically (so, not just relying on the generic media keys) as well as direct shortcuts to certain channels

**Comment by TimBurris** [comment]
yes it absolutely lends itself to additional scripting.  Ultimately, user and system interactions are handled in WPF (.NET), then those keystrokes, system events, etc are used to manipulate the DOM of the SiriusXM website.

Linking to a specific channel would likely be just redirecting to the channel's url (if each channel has a pre-determined url)

I haven't used streamdeck before, but i suspect the inputs from streamdeck can be captured via an api or even windows events, once you capture it, you can manipulate the siriusxm DOM as needed.  

P.S.
i'm not sure your familiarity with DOM and WPF, I don't want to assume nor insult!

**Comment by jsblake** [comment]
Streamdeck can either send keypresses (and combos) or run external scripts (like powershell)

There's also an API that could get a little deeper 

I don't have a lot of experience with DOM or WPF, but I'm going to start doing some digging, thanks for making this project open source
